src/strtree.c

```diff
@@ -197,6 +199,38 @@ static PyObject* STRtree_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
     counter++;
   }
 
+  // A dummy query to trigger the build of the tree
+  tree_geom_vec_t query_geoms;
+  kv_init(query_geoms);
+
+  GEOSCoordSequence* coord_seq;
+  GEOSGeometry* dummy;
+
+  coord_seq = GEOSCoordSeq_create_r(ctx, 1, 2);
+  if (coord_seq == NULL) {
+    GEOSSTRtree_destroy_r(ctx, tree);
+    GEOS_FINISH;
+    return NULL;
+  }
+  for (int j = 0; j < 2; j++) {
+    if (!GEOSCoordSeq_setOrdinate_r(ctx, coord_seq, 0, j, 0.0)) {
+      GEOSCoordSeq_destroy_r(ctx, coord_seq);
+      GEOSSTRtree_destroy_r(ctx, tree);
+      GEOS_FINISH;
+      return NULL;
+    }
+  }
+  dummy = GEOSGeom_createPoint_r(ctx, coord_seq);
+  if (dummy == NULL) {
+    GEOSCoordSeq_destroy_r(ctx, coord_seq);
+    GEOSSTRtree_destroy_r(ctx, tree);
+    GEOS_FINISH;
+    return NULL;
+  }
```

Comment:
@caspervdw do you know of a less verbose way to create a simple point?

Reply:
`GEOSGeom_createPointFromXY` was introduced in GEOS 3.8.  Though it adds a bit of code, you could use this in a `#if GEOS_SINCE_3_8_0` block with a fallback to the coordinate sequence varient above, so that eventually when that is the only supported version, this could be simplified.

Reply:
Maybe use an empty point for the query? Creating one of those is certainly less verbose, but I am not sure whether a query with an empty point would trigger the tree to be built.

Reply:
Looking in the GEOS code, I think it currently doesn't check if the envelope is empty before checking if the tree is built (https://github.com/libgeos/geos/blob/edd209ae551bda9cdb6188230a1861937153ce79/include/geos/index/strtree/TemplateSTRtree.h#L243-L253, it's only the `root->boundsIntersect(queryEnv)` that would never return true for an empty point), but not sure to what extent this should be considered an implementation detail.